From b74ec43a299ccf82c74decffc3d5508c37feb2ef Mon Sep 17 00:00:00 2001
From: David Cohen <david.a.cohen@intel.com>
Date: Thu, 19 Mar 2015 17:06:59 +0100
Subject: [PATCH 214/639] dwc3: increase autosuspend timeout

Current autosuspend timeout is too short. The autosuspend in this case is
useful to avoid a power cycle it an user is unplugging/plugging usb cable in a
short moment. But since current time is 0.5s, it's unlikely the autosuspend
will ever bring any positive side effect.

This patch increases the deadline to 10s to be more realistic.

Change-Id: Id669fd3785f3550bdf2abe0934c2ebcb2152864b
Signed-off-by: David Cohen <david.a.cohen@intle.com>
(cherry picked from commit e945d8ab4a64840c2cfa94dfd31f3671675a6d4e)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/usb/dwc3/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 65e4964..23db3a3 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -776,7 +776,7 @@ static int dwc3_probe(struct platform_device *pdev)
 	atomic_set(&dwc->suspend_depth, 0);
 
 	if (dwc->runtime_suspend) {
-		pm_runtime_set_autosuspend_delay(dev, 500);
+		pm_runtime_set_autosuspend_delay(dev, 10000);
 		pm_runtime_use_autosuspend(dev);
 		pm_runtime_mark_last_busy(dev);
 		pm_runtime_put_autosuspend(dev);
-- 
2.10.1

